### PR TITLE
Fix test create and restore searchable snapshot

### PR DESCRIPTION
**Pull request by idegtiarenko**
Related to: #89687

### CODE REVIEW

File: x-pack/plugin/searchable-snapshots/src/internalClusterTest/java/org/elasticsearch/xpack/searchablesnapshots/BaseSearchableSnapshotsIntegTestCase.java [hunk 1]
```diff
@@ -246,18 +247,18 @@ protected void assertShardFolders(String indexName, boolean snapshotDirectory) t
...
                 assertThat(
-                    snapshotDirectory ? "Snapshot directory doesn't exist" : "Snapshot directory shouldn't exist",
-                    snapshotDirectory,
-                    not(indexExists)
+                    snapshotDirectory ? "Index file should not exist" : "Index file should exist",
+                    indexExists,
+                    not(snapshotDirectory)
                 );
-                assertTrue("Translog doesn't exist", translogExists);
+                assertThat("Translog should exist", translogExists, is(true));
```
Comment (idegtiarenko) [inline x-pack/plugin/searchable-snapshots/src/internalClusterTest/java/org/elasticsearch/xpack/searchablesnapshots/BaseSearchableSnapshotsIntegTestCase.java:254]: Above rewords assertions messages to remove negation

File: x-pack/plugin/searchable-snapshots/src/internalClusterTest/java/org/elasticsearch/xpack/searchablesnapshots/SearchableSnapshotsIntegTests.java [hunk 2]
```diff
@@ -254,7 +255,7 @@ public void testCreateAndRestoreSearchableSnapshot() throws Exception {
         assertSearchableSnapshotStats(restoredIndexName, cacheEnabled, nonCachedExtensions);
 
         ensureGreen(restoredIndexName);
-        assertBusy(() -> assertShardFolders(restoredIndexName, true));
+        assertBusy(() -> assertShardFolders(restoredIndexName, true), 30, TimeUnit.SECONDS);
```
Comment (idegtiarenko) [inline x-pack/plugin/searchable-snapshots/src/internalClusterTest/java/org/elasticsearch/xpack/searchablesnapshots/SearchableSnapshotsIntegTests.java:258]: This is consistent with another change: https://github.com/elastic/elasticsearch/pull/84942

Comment (idegtiarenko) [reply x-pack/plugin/searchable-snapshots/src/internalClusterTest/java/org/elasticsearch/xpack/searchablesnapshots/SearchableSnapshotsIntegTests.java:258]: It looks like this reproduces only if we close the index (eg index files are kept on the disk) **and** we reuse the same index name for the mounted index. This is likely resulting in files not cleaned in time.

**Review by tlrx — APPROVED** [review]
LGTM

**Pull request merged**